crates/tauri-plugin-holochain-service-consumer/src/mobile.rs

```diff
@@ -30,37 +30,42 @@ pub fn init<R: Runtime, C: DeserializeOwned>(
 pub struct HolochainServiceConsumer<R: Runtime>(pub PluginHandle<R>);
 
 impl<R: Runtime> HolochainServiceConsumer<R> {
+    /// Runs the entire process to setup a holochain app
+    ///
+    /// 1. Connect to holochain service
+    /// 2. Check if app is already installed
+    /// 3. If not installed, then install then app
+    /// 4. Optionally, enable the app
+    /// 5. Ensure there is an app websocket and return its authentication config
+    ///
+    /// This can be safely called whether or not the app is installed.
+    pub fn setup_app(&self, config: SetupAppConfig) -> crate::Result<AppAuthFfi> {
+        Ok(self.0.run_mobile_plugin::<AppAuthFfi>("setupApp", config)?)
+    }
+
     /// Build a window that opens the main UI for your Tauri app.
     /// This is equivalent to creating a window with `WebviewUrl::App(PathBuf::from("index.html"))`.
     ///
-    /// * `app_id` - The `app_id` for the app.
-    ///   If an app with this `app_id` is not installed in the Holochain conductor running in the Android Service Runtime,
-    ///   then this happ bundle will be installed with the provided `app_id` and `network_seed`.
-    ///   Afterwards, the window will be setup so a holochain client can connect to that app.
-    /// * `happ_bundle_bytes` - The raw bytes of the `.happ` file.
-    /// * `network_seed` - The network seed to include in the `InstallAppPayload`.
-    /// * `enable_app` - If `true`, enable_app will be called for the app, after it has been installed successfully.
+    /// * `config`: The App Websocket config to inject into the webview
     pub fn main_window_builder(
         &self,
         app_id: String,
-        happ_bundle_bytes: Bytes,
-        network_seed: String,
-        enable_app: bool,
+        auth: Option<AppAuthFfi>,
     ) -> tauri::Result<WebviewWindowBuilder<R, AppHandle<R>>> {
         let label = "main";
-        let window_builder =
+        let mut window_builder =
             WebviewWindowBuilder::new(self.0.app(), label, WebviewUrl::App("".into()))
-                .initialization_script(include_str!("../dist-js/holochain-env/index.min.js"))
-                .initialization_script(
-                    format!(
-                        r#"setupApp("{}", {:?}, "{}", {});"#,
-                        app_id,
-                        happ_bundle_bytes.deref(),
-                        network_seed,
-                        enable_app,
-                    )
-                    .as_str(),
-                );
+                .initialization_script(include_str!("../dist-js/holochain-env/index.min.js"));
+
+        if let Some(auth) = auth {
+            window_builder = window_builder.initialization_script(
+                format!(
+                    r#"injectHolochainClientEnv("{}", {}, {:?});"#,
```

Comment:
Does it not need to be `window.injectHolochainClientEnv(...)`? I'm assuming not as you've probably tested it but I would have expected it from looking at the code of `holochain/env/index.ts`.

Reply:
It does not but might as well make it explicit: https://github.com/holochain/android-service-runtime/pull/78/commits/8106b68c5a2b609a03fabefa68bc7178de313a2e